components/engine_rocks/src/write_batch.rs

```diff
@@ -109,6 +113,10 @@ impl engine_traits::WriteBatch<RocksEngine> for RocksWriteBatch {
     fn rollback_to_save_point(&mut self) -> Result<()> {
         self.wb.rollback_to_save_point().map_err(Error::Engine)
     }
+
+    fn merge(&mut self, src: &Self) {
+        self.merge(src);
```

Comment:
How about moving L68~L70 here? This implementation makes it look like indefinitely recursive calls.